tests/test_branch.py

```diff
@@ -0,0 +1,168 @@
+#!/usr/bin/env python3
+from contextlib import contextmanager
+from datetime import datetime, timedelta
+import os
+from pathlib import Path
+from tempfile import TemporaryDirectory
+import subprocess
+import unittest.mock
+from typing import Any, BinaryIO, Iterator
+from nix_update.version import fetch_latest_version
+from nix_update.version.version import VersionPreference
+import conftest
+
+TEST_ROOT = Path(__file__).parent.resolve()
+
+
+def create_dummy_commit(tmp: str, date: datetime) -> None:
+    datestr = date.strftime("%FT%TZ")
+    with open(Path(tmp) / "date.txt", "w", encoding="utf-8") as fp:
+        fp.write(datestr)
+        fp.write("\n")
+    subprocess.run(["git", "-C", tmp, "add", "date.txt"], check=True)
+    env = os.environ.copy()
+    env["GIT_AUTHOR_DATE"] = datestr
+    env["GIT_COMMITTER_DATE"] = datestr
+    subprocess.run(
+        [
+            "git",
+            "-C",
+            tmp,
+            "-c",
+            "user.name=dummy",
+            "-c",
+            "user.email=dummy@localhost",
+            "commit",
+            "--quiet",
+            "--message=some message",
+        ],
+        env=env,
+        check=True,
+    )
+
+
+atom_feed_log_format = """\
+  <entry>
+    <link href="/%H"/>
+    <updated>%cd</updated>
+  </entry>"""
+
+
+def create_dummy_commit_feed(tmp: str, branch: str) -> None:
+    with open(Path(tmp) / f"{branch}.atom", "wb") as feed:
+        feed.write(
+            b"""\
+<?xml version="1.0" encoding="UTF-8"?>
+<feed xmlns="http://www.w3.org/2005/Atom">
+"""
+        )
+        feed.flush()
+        env = os.environ.copy()
+        env["TZ"] = "UTC"
+        subprocess.run(
+            [
+                "git",
+                "-C",
+                tmp,
+                "log",
+                "--date=format-local:%FT%TZ",
+                f"--format=format:{atom_feed_log_format}",
+                "--max-count=20",
+                branch,
+            ],
+            env=env,
+            stdout=feed,
+            check=True,
+        )
+        feed.write(
+            b"""
+</feed>
+"""
+        )
```

Comment:
I don't think we need to rebuild a dummy RSS feed. We could just save one and or create one by hand for testing and use that in the tests. I don't think GitHub will change the format anytime soon and that saves a lot of unnecessary complexity.

Reply:
I quite dislike git submodules but since another repo is needed for the test I decided to create it and its feeds.

Reply:
Do we even need a repo in the first place? We can just use mock data here, right?

Reply:
A repo is needed for testing `get_latest_snapshot_from_git`.

Reply:
Hmmm, I am not convinced that we actually need that function. Do git clone have no/higher rate limit compared to the tarball download? I think cloning adds so much complexity which we should try to avoid. Also I am not sure if we need all the special date compare logic. So far I never used nix-update for anything but to get the latest and greatest and did everything else by hand. Am I missing something here?

Reply:
I would use `--before` by default, because, as I said, I would like the dates to be unique.

I will clean up the commits now and move everything relating to `get_latest_snapshot_from_git` and `--before` to [a separate branch](https://github.com/schnusch/nix-update/compare/unstable..before).